haystack/nodes/answer_generator/base.py

```diff
@@ -72,12 +72,13 @@ def _create_answers(self, generated_answers: List[str], documents: List[Document
             answers.append(
                 Answer(
                     answer=generated_answer,
+                    document_ids=flat_docs_dict["id"],
                     type="generative",
                     meta={
-                        "doc_ids": flat_docs_dict["id"],
                         "doc_scores": flat_docs_dict["score"],
                         "content": flat_docs_dict["content"],
                         "titles": [d.get("name", "") for d in flat_docs_dict["meta"]],
+                        "doc_metas": flat_docs_dict["meta"],
```

Comment:
We need to add this for `eval` to have access to the meta of the documents to resolve the `custom_document_id`. This was missing till now for generative QA completely. 
Note that the Reader copies the document meta right into the answer meta (for extractive QA).